expr/func_summarize_test.go

```diff
@@ -665,6 +665,78 @@ func TestSummarizeAlignToFrom(t *testing.T) {
 	testSummarize("AlignToFrom", input, outputMax[1], "45s", "max", true, t)
 }
 
+func TestSummarizeLargeIntervalTimestamps(t *testing.T) {
+
+	// This test is specifically testing the timestamp behavior, so we don't need real values.
+	// However, we do need a lot of datapoints to trigger the bug we are regression testing against.
+	interval1Day := uint32(24 * 60 * 60)
+	numTimestamps := uint32(2000)
+	startTime := uint32(1464637518)
+	endTime := startTime + numTimestamps*interval1Day
+
+	inputDps := make([]schema.Point, 0, numTimestamps)
+
+	for i := uint32(0); i < numTimestamps; i++ {
+		inputDps = append(inputDps, schema.Point{Val: 0, Ts: startTime + i*interval1Day})
+	}
+
+	outputInterval := 365 * interval1Day
+
+	var unaligned365dsum = []schema.Point{
+		{Val: 0, Ts: 1450656000},
+		{Val: 0, Ts: 1482192000},
+		{Val: 0, Ts: 1513728000},
+		{Val: 0, Ts: 1545264000},
+		{Val: 0, Ts: 1576800000},
```

Comment:
what's the relation between these ts? can they be set from code so it's more obvious?

Reply:
These are the expected outcome. Setting from code would be partially duplicating the logic in the function under test, which I try to avoid in test code.

Reply:
(These are the results as graphite reported them)

Reply:
i agree we shouldn't use advanced code that is similar to or shared with the code under test,
but now this is pretty unreadable: i have to pull out a calculator to find out they are apart by 2592000 , aka 30*24*3600, and to confirm that the first ts is the last interval boundary before the startTime.
I think trivial, obviously-correct solves this problem. Thus I would recommend something like:

```
   start := startTime - (startTime % outputInterval)
        var unalignedExpected = []schema.Point{
                {Val: 0, Ts: start},
                {Val: 0, Ts: start + outputInterval},
                {Val: 0, Ts: start + 2*outputInterval},
                {Val: 0, Ts: start + 3*outputInterval},
        }

```